packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.types.ts

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import { IStyle, ITheme } from '../../Styling';
+import { IStyleFunctionOrObject } from '../../Utilities';
+
+export interface IShimmer {}
+
+/**
+ * Shimmer component props.
+ */
+export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
+  /**
+   * Optional callback to access the IShimmer interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IShimmer | null) => void;
+
+  /**
+   * Sets the width of the shimmer wave wrapper in percentages relative to the containig parent element.
+   * @default 100%
+   */
+  widthInPercentage?: number;
+
+  /**
+   * Sets the width of the shimmer wave wrapper to an exact value in pixels.
+   * If none of the widths provided, it defaults to 100%.
+   */
+  widthInPixel?: number;
+
+  /**
+   * Controls when the shimmer is swapped with actual data through an animated transition.
+   * @default false
+   */
+  isDataLoaded?: boolean;
+
+  /**
+   * Elements to render in one line of the Shimmer.
+   */
+  shimmerElements?: IShimmerElement[];
```

Comment:
Is this like for showing gray boxes and such? If this is for one line, does that mean that every line in the Shimmer is identical?

Reply:
Shimmer is actually the one providing the running animation. `shimmerElements` and `custom Elements` are just transparent divs (except the gap) sitting on top of the animation and covering with borders the areas you don't want to see. This allowed us to have a visual effect of the shimmering wave going from the one side to another of the provided overall width of the Shimmer in a synchronous fashion, no matter how manny elements we have in that line.